core/common/src/main/java/alluxio/util/ShellUtils.java

```diff
@@ -193,6 +194,101 @@ private String run() throws ExitCodeException, IOException {
         process.destroy();
       }
     }
+
+    /**
+     * Runs a command and returns its stdout, stderr and exit code.
+     * No matter it succeeds or not.
+     *
+     * @return {@link CommandReturn} object representation of stdout, stderr and exit code
+     */
+    protected CommandReturn runTolerateFailure() throws IOException {
```

Comment:
This method is similar to `run()`.

The differences are:
1. Not redirecting stderr to stdout. 
2. Keeping exitCode, stderr and stdout separated and wrapped in an object CommandReturn.
3. Not throwing exception when exitcode is non-zero. Return all information instead.